lib/middleware/notify.js

```diff
@@ -0,0 +1,30 @@
+var updateNotifier = require("update-notifier");
+var helpers        = require("./util/helpers");
+var isNPM          = require("is-npm");
+var pkg            = require(__dirname + "./../../package.json");
+
+module.exports = function(req, next){
+  // Checks for available update and returns an instance
+  var notifier = updateNotifier({
+    // updateCheckInterval: 1000, // For testing
+    pkg: pkg
+});
+
+  if (!process.stdout.isTTY || isNPM || !notifier.update) {
```

Comment:
Reverse this `if` for clarity's sake. When the situation is special, do something special.

``` javascript
if (special && case) {
  takeSpecialAction();
}
// always take this action
next();
```